core/tests/protractor_utils/forms.js

```diff
@@ -581,51 +585,52 @@ var CodeMirrorChecker = function(elem, codeMirrorPaneToScroll) {
     // if scrollTo is too large.
-    browser.executeScript(
+    await browser.executeScript(
       '$(\'.CodeMirror-vscrollbar\').' + codeMirrorPaneToScroll +
       '().scrollTop(' + String(scrollTo) + ');');
-    elem.all(by.xpath('./div')).map(function(lineElement) {
-      return lineElement.element(by.css('.CodeMirror-linenumber')).getText()
-        .then(function(lineNumber) {
-          // Note: the last line in codemirror will have an empty string for
-          // line number and for text. This is to skip that line.
-          if (lineNumber === '') {
-            return lineNumber;
-          }
-          if (!compareDict.hasOwnProperty(lineNumber)) {
-            throw new Error('Line ' + lineNumber + ' not found in CodeMirror');
-          }
-          expect(lineElement.element(by.xpath('./pre')).getText())
-            .toEqual(compareDict[lineNumber].text);
-          expect(
-            lineElement.element(
-              by.css('.CodeMirror-linebackground')).isPresent())
-            .toEqual(compareDict[lineNumber].highlighted);
-          compareDict[lineNumber].checked = true;
+    var lineNumbers = await elem.all(by.xpath('./div')).map(
+      async function(lineElement) {
+        var lineNumber = await lineElement.element(
+          by.css('.CodeMirror-linenumber')).getText();
+        // Note: the last line in codemirror will have an empty string for
+        // line number and for text. This is to skip that line.
+        if (lineNumber === '') {
           return lineNumber;
-        });
-    }).then(function(lineNumbers) {
-      var largestLineNumber = lineNumbers[lineNumbers.length - 1];
-      if (largestLineNumber !== currentLineNumber) {
-        _compareTextAndHighlightingFromLine(
-          largestLineNumber,
-          scrollTo + CODEMIRROR_SCROLL_AMOUNT_IN_PIXELS,
-          compareDict);
-      } else {
-        for (var lineNumber in compareDict) {
-          if (compareDict[lineNumber].checked !== true) {
-            throw new Error(
-              'Expected line ' + lineNumber + ': \'' +
-              compareDict[lineNumber].text + '\' to be found in CodeMirror');
-          }
         }
+        if (!compareDict.hasOwnProperty(lineNumber)) {
+          throw new Error('Line ' + lineNumber + ' not found in CodeMirror');
+        }
+        expect(await lineElement.element(by.xpath('./pre')).getText())
+          .toEqual(compareDict[lineNumber].text);
+        expect(
+          await lineElement.element(
+            by.css('.CodeMirror-linebackground')).isPresent())
+          .toEqual(compareDict[lineNumber].highlighted);
+        compareDict[lineNumber].checked = true;
+        return lineNumber;
       }
-    });
+    );
+    var largestLineNumber = lineNumbers[lineNumbers.length - 1];
+    if (largestLineNumber !== currentLineNumber) {
+      await _compareTextAndHighlightingFromLine(
+        largestLineNumber,
+        scrollTo + CODEMIRROR_SCROLL_AMOUNT_IN_PIXELS,
+        compareDicti
```

Comment:
@U8NWXD looks like a typo.. Please make sure to test code that you push.

Reply:
Thanks for flagging this!